benchmark/buffers/buffer-bytelength-utf8.js

```diff
@@ -0,0 +1,42 @@
+var common = require('../common');
+
+var bench = common.createBenchmark(main, {
+    len: [4, 16, 64, 256, 1024],
+    n: [1e6]
+});
+
+// 16 chars of each byte length
+var chars = {
+  0: 'hello brendan!!!', // 1 byte
+  1: 'ΰαβγδεζηθικλμνξο', // 2 bytes
+  2: '挰挱挲挳挴挵挶挷挸挹挺挻挼挽挾挿', // 3 bytes
+  3: '𠜎𠜱𠝹𠱓𠱸𠲖𠳏𠳕𠴕𠵼𠵿𠸎𠸏𠹷𠺝𠺢' // 4 bytes
+};
+
+function main(conf) {
+  var n = conf.n | 0;
+  var len = conf.len | 0;
+
+  // make a string with them all mixed together
+  var str = buildStr(len);
+
+  bench.start();
+  for (var i = 0; i < n; i++) {
+    Buffer.byteLength(chars[0], 'utf8');
```

Comment:
The meaning of the guards is not to be a test but to make the benchmark less susceptible to unrealistic optimizations like licm and dce. There is no reason to use assert and the checks must be done in the loop, otherwise they are ineffective.